### ISSUE TITLE
OPTUNA Trials

### ISSUE THREAD

**Issue by manza-ari** [optuna/optuna#3997]
### What is an issue?

Hi, 
Thank you so much for your question. 
I have some questions about OPTUNA-Trials, what it is doing exactly and why it is so time-consuming, I have three different types of objective functions and I am running each of them for 50-OPTUNA Trials. I want to know the insight of OPTUNA for each Trial for each objective.  

`Objective 1
optuna_initial_lr = optuna_trial.suggest_float('LearningRate-for-Random-ImageNet', 1e-3, 1e-1, log=True)

Objective 2
optim_epochs = optuna_trial.suggest_int('no_epochs', 100, 300, step = 50) 

Objective 3
optuna_name = optuna_trial.suggest_categorical('optimizer', ['Adam', 'SGD', 'RMSprop', 'SAMSGD']) 

                if optuna_name == 'Adam':
                        criterion      = nn.CrossEntropyLoss(reduction='none')
                        optim_backbone = optim.Adam(models.parameters(), weight_decay=WDECAY)
                if optuna_name == 'SGD':
                        criterion      = nn.CrossEntropyLoss(reduction='none')
                        optim_backbone = optim.SGD(models.parameters(), lr=LR, momentum=MOMENTUM, weight_decay=WDECAY)
                if optuna_name == 'RMSprop':
                        criterion      = nn.CrossEntropyLoss(reduction='none')
                        optim_backbone = optim.RMSprop(models.parameters(), momentum=MOMENTUM, weight_decay=WDECAY)
                if optuna_name == 'SAMSGD':
                        criterion      = nn.CrossEntropyLoss(reduction='mean')
                        base_optimizer = torch.optim.SGD
                        optim_backbone = SAM(models.parameters(), base_optimizer, lr=LR, momentum=MOMENTUM, weight_decay=WDECAY)

  
    import optuna
    from optuna_custom_visualization import *
    import matplotlib.pyplot as plt
    

    study = optuna.create_study(direction="maximize")  # Create a new study.
    study.optimize(Objective(), n_trials=1)  # Invoke optimization of the objective function.

    #Slice Plot
    
    plot_slice = optuna.visualization.plot_slice(study,  target_name = 'Test Accuracy')
    plot_slice.write_image("SlicePlot-Random-CIFAR100-OptimalOptimizer.png", width = 1635, height = 923)

                
`
